File: adapters/runtime/common/dom-adapter/audioContext/AudioDestinationNode.js
import AudioNode from "./AudioNode";

class AudioDestinationNode extends AudioNode {
    /**
     *
     * @param context {BaseAudioContext}
     */
    constructor(context) {
        super(context);

        this.maxChannelCount = 2; // 设备可以处理的最大通道数
    }
}

export default AudioDestinationNode;

File: native/cocos/audio/graph_based/AudioDestinationNode.cpp
#include "audio/graph_based/AudioDestinationNode.h"
namespace cc {
void AudioDestinationNode::setMaxChannelCount(uint32_t channelCount) {
    _maxChannelCount = channelCount;
    // TODO: set lab::AudioDestinationNode max channel count;
}
AudioDestinationNode* AudioDestinationNode::createDestination(BaseAudioContext* ctx, std::shared_ptr<lab::AudioNode> node) {
    return new AudioDestinationNode(ctx, node);
}
AudioDestinationNode::AudioDestinationNode(BaseAudioContext* ctx, std::shared_ptr<lab::AudioNode> node) : AudioNode(ctx){
    _node = node;
    _maxChannelCount = node->channelCount();
};
}
